### PR TITLE
feat(CreateEdit): Added character limit for short description #293

### PR DESCRIPTION
**Pull request by wski**
@arw180 @blaino @clarkjefcoat can you review

### CODE REVIEW

File: app/js/components/createEdit/form/InputMixin.jsx [hunk 1]
```diff
@@ -13,13 +13,16 @@ var InputMixin = {
 
     render: function () {
         var labelClasses = classSet({ 'input-optional': this.props.optional });
-
+        var CharLimit = (this.props.charLimit && this.props.value) ? CharLimit = `${(this.props.charLimit - this.props.value.length > 0) ? this.props.charLimit - this.props.value.length : 0 }/${this.props.charLimit} characters remaining.` :  `${this.props.charLimit}/${this.props.charLimit} characters remaining.`;
```
Comment (blaino) [inline app/js/components/createEdit/form/InputMixin.jsx]: too long.

File: app/js/components/createEdit/form/InputMixin.jsx [hunk 2]
```diff
@@ -13,13 +13,20 @@ var InputMixin = {
 
     render: function () {
         var labelClasses = classSet({ 'input-optional': this.props.optional });
-
+        // if we have a charlimit and value is not undefined display the characters remaining.
+        var CharLimit = (this.props.charLimit && this.props.value) ?
```
Comment (blaino) [inline app/js/components/createEdit/form/InputMixin.jsx]: This is fine, but my personal preference is to do the math first to get the number of remaining characters and just build the string after that.

File: app/js/components/createEdit/form/InputMixin.jsx [hunk 3]
```diff
@@ -13,13 +13,20 @@ var InputMixin = {
...
+            `${this.props.charLimit - this.props.value.length}/${this.props.charLimit} characters remaining`:
+            // if we do have a character limit, but value is undefined set it to the char limit max remaining
+            `${this.props.charLimit}/${this.props.charLimit} characters remaining`;
```
Comment (blaino) [inline app/js/components/createEdit/form/InputMixin.jsx]: Isn't this just "1 character remaining?" this.props.charLimit/this.props.charLimit = 1?

**Comment by blaino** [comment]
:+1: 

I pulled it down and ran it locally - works as expected. 

@arw180, next review please?

**Comment by arw180** [comment]
Please `rebase -i` these 5 commits down into 1. Also, use `feat` instead of `chore` for the commit type so that it shows up in the changelog. Further review pending ...

**Comment by arw180** [comment]
This did not work quite as expected. When I type one char, it says 99 characters remaining (good so far), but when I type the second character, it still says 99 characters remaining. Typing a third character results in "98 characters remaining", and then it counts down from there.

**Comment by wski** [comment]
for whatever reason using state instead of props seems to have rectified the issue fore me (could only reproduce in Safari, so tested there)

**Comment by arw180** [comment]
After the last change it appears to work as expected, but your rebase wasn't done properly as there are still two commits

**Comment by wski** [comment]
I am aware, I will reply here when I am ready for you to check

**Comment by wski** [comment]
rebased to one commit

**Comment by arw180** [comment]
:+1:

**Pull request merged**